js/ui/components/clubhouse.js

```diff
@@ -55,6 +55,82 @@ const ClubhouseIface =
   </interface> \
 </node>';
 
+var ClubhouseWindowTracker = new Lang.Class({
+    Name: 'ClubhouseWindowTracker',
+
+    _init: function() {
+        this.parent();
```

Comment:
I will keep it since @GeorgesStavracas has asked for it to be a GObject. I'd rather skip the extra code if using `addSignalMethods` works fine, but if this is more appropriate in the Shell, then I will do it.